lms/static/sass/shared/_course_object.scss

```diff
@@ -7,33 +7,44 @@
       width: flex-grid(4);
       @include margin-right(flex-gutter());
       float: left;
-
-      &:nth-child(3n+3) {
-        @include margin-right(0);
-      }
     }
 
     .courses-listing {
+      @include outer-container;
       @include clearfix();
       margin: 0;
       padding: 0;
       list-style: none;
 
       .courses-listing-item {
-        width: flex-grid(4);
-        @include margin-right(flex-gutter());
-        @include float(left);
+        height: 360px;
+
+        @include media($mobile) {
+          @include span-columns(4);
+          margin: 15px 0px;
+        }
+
+        @include media($tablet) {
+          @include span-columns(4);
+          margin: 15px 3px 15px;
+        }
 
-        &:nth-child(3n+3) {
-          @include margin-right(0);
+        @include media($desktop) {
+          @include span-columns(4);
+          margin: 15px 5px 15px;
         }
+
+        @include media($xl-desktop) {
+          @include span-columns(3);
+          margin: 15px 7px 15px;
```

Comment:
Redundant margin-top and margin-bottom rules here (your base rules should be inherited and cover this case). Please remove these.